song-server/src/main/java/bio/overture/song/server/service/StorageService.java

```diff
@@ -58,18 +58,19 @@
   @NonNull private final RetryTemplate retryTemplate;
   @NonNull private final String storageUrl;
   @NonNull private final ValidationService validationService;
+  @NonNull private final String downloadAccessToken;
 
   @SneakyThrows
   public boolean isObjectExist(@NonNull String accessToken, @NonNull String objectId) {
     return doGetBoolean(accessToken, getObjectExistsUrl(objectId));
   }
 
   @SneakyThrows
-  public StorageObject downloadObject(@NonNull String accessToken, @NonNull String objectId){
-    val objectExists = isObjectExist(accessToken, objectId);
+  public StorageObject downloadObject(@NonNull String objectId){
+    val objectExists = isObjectExist(downloadAccessToken, objectId);
```

Comment:
just remembered, this `isObjectExist` method does a `GET storage.cancercollaboratory.org/upload/<objectid>`  call and receives a true or false if the object exists. This means it requires an `upload` scope to make this call. The application level score accesstoken should have both upload and download scope, to allow for the isObjectExist check to work, as well as download from the server.